### PR TITLE
Add to_line_stream/1

### LINKED ISSUE #57: Transform stream to line based one before trying to parse
Sometimes streams are not yet line based (e.g. when streaming over http). I'm wondering if it makes sense for nimble_csv to optionally deal with such by doing something akin to this before parsing the data:

```elixir
chunk_fun = fn element, acc ->
  parts = String.split(element, "\n")

  case List.pop_at(parts, -1) do
      {nil, []} -> {[], acc}
      {new_acc, []} -> {[], new_acc}
      {new_acc, [h | t]} -> {[acc <> h | t], new_acc}
  end
end
after_fun = fn
  "" -> []
  acc -> [acc]
end

[
    "abc",
    "def\n",
    "abc\ndef",
    "abc\ndef\nmore",
    "\nok"
]
|> Stream.transform(fn -> "" end, chunk_fun, after_fun)
|> Enum.into([])
|> IO.inspect()
# ["abcdef", "abc", "defabc", "def", "more"]
```

### PR DESCRIPTION
**Pull request by LostKobrakai**
Closes #57

### CODE REVIEW

**Comment by LostKobrakai** [comment]
So seems like we're off looking for a <1.5 way for `chunk_while`. I tried using `Stream.transform/4`, but the after fun sadly cannot make the final acc be considered an element of the stream.

File: lib/nimble_csv.ex [hunk 1]
```diff
@@ -326,6 +335,53 @@ defmodule NimbleCSV do
...
+      def to_line_stream(stream) do
+        newline = :binary.compile_pattern(@encoded_newlines)
+
+        stream
+        |> Stream.chunk_while(
+          "",
+          &to_line_stream_chunk_fun(&1, &2, newline),
+          &to_line_stream_after_fun/1
+        )
+        |> Stream.flat_map(&Function.identity/1)
```
Comment (josevalim) [inline lib/nimble_csv.ex]:
```suggestion
        |> Stream.concat()
```

File: lib/nimble_csv.ex [hunk 2]
```diff
@@ -326,6 +335,55 @@ defmodule NimbleCSV do
...
+      def to_line_stream(stream) do
+        newline = :binary.compile_pattern(@encoded_newlines)
+
+        stream
+        |> Stream.chunk_while(
+          "",
+          &to_line_stream_chunk_fun(&1, &2, newline),
+          &to_line_stream_after_fun/1
+        )
+        |> Stream.flat_map(&identity/1)
```
Comment (josevalim) [inline lib/nimble_csv.ex:347]: I think `Stream.concat()` is available here too.

File: lib/nimble_csv.ex [hunk 3]
```diff
@@ -326,6 +335,55 @@ defmodule NimbleCSV do
         |> parse_enumerable(opts)
       end
 
+      def to_line_stream(stream) do
+        newline = :binary.compile_pattern(@encoded_newlines)
+
+        stream
+        |> Stream.chunk_while(
```
Comment (josevalim) [inline lib/nimble_csv.ex:342]: We don't have this available on v1.3 but we do have it on v1.5. It is fine to bump the requirement to ~> 1.5. Just please remember to upgrade mix.exs and CI files. :)

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**